apps/synapse-portal-framework/src/Portal.tsx

```diff
@@ -1,55 +1,33 @@
-import { BrowserRouter, Switch } from 'react-router-dom'
+import { createTheme, ThemeProvider } from '@mui/material'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 import React, { useMemo } from 'react'
-import Footer from './components/Footer'
-import AppInitializer from './components/AppInitializer'
-import Navbar from './components/Navbar'
 import { CookiesProvider } from 'react-cookie'
-import {
-  CookiesNotification,
-  SynapseTheme,
-  SynapseToastContainer,
-  defaultQueryClientConfig,
-} from 'synapse-react-client'
+import { createBrowserRouter, RouterProvider } from 'react-router-dom'
+import { defaultQueryClientConfig, SynapseTheme } from 'synapse-react-client'
 import { LogInDialogContextProvider } from './components/LogInDialogContext'
-import { createTheme, ThemeProvider } from '@mui/material'
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
-import { PortalProps } from './components/PortalProps'
 import { PortalContextProvider } from './components/PortalContext'
+import { PortalProps } from './components/PortalProps'
 
-const RouteResolver = React.lazy(() => import('./components/RouteResolver'))
 const queryClient = new QueryClient(defaultQueryClientConfig)
 
 function Portal(props: PortalProps) {
   const { palette, ...context } = props
+  const router = createBrowserRouter(props.routeConfig)
   const theme = useMemo(
     () => createTheme(SynapseTheme.mergeTheme({ palette })),
     [palette],
   )
+
   return (
     <PortalContextProvider value={context}>
       <CookiesProvider>
-        <BrowserRouter>
-          <LogInDialogContextProvider>
-            <ThemeProvider theme={theme}>
-              <QueryClientProvider client={queryClient}>
-                <AppInitializer>
-                  <SynapseToastContainer />
-                  <Navbar />
-                  <CookiesNotification />
-                  <main className="main">
-                    {/* all the content below */}
-                    <React.Suspense fallback={<div />}>
-                      <Switch>
-                        <RouteResolver />
-                      </Switch>
-                    </React.Suspense>
-                  </main>
-                  <Footer />
-                </AppInitializer>
-              </QueryClientProvider>
-            </ThemeProvider>
-          </LogInDialogContextProvider>
-        </BrowserRouter>
+        <LogInDialogContextProvider>
+          <ThemeProvider theme={theme}>
+            <QueryClientProvider client={queryClient}>
+              <RouterProvider router={router} />
```

Comment:
Components containing rendered content were moved to App.tsx, and rendering content is now entirely handled by React Router via `RouterProvider`